commands/core/upgrade.go

```diff
@@ -25,12 +25,31 @@ import (
 )
 
 // PlatformUpgrade FIXMEDOC
-func PlatformUpgrade(ctx context.Context, req *rpc.PlatformUpgradeRequest,
-	downloadCB rpc.DownloadProgressCB, taskCB rpc.TaskProgressCB) (*rpc.PlatformUpgradeResponse, error) {
+func PlatformUpgrade(
+	ctx context.Context,
+	req *rpc.PlatformUpgradeRequest,
+	downloadCB rpc.DownloadProgressCB,
+	taskCB rpc.TaskProgressCB,
+) (*rpc.PlatformUpgradeResponse, error) {
+	if err := platformUpgrade(ctx, req, downloadCB, taskCB); err != nil {
+		return nil, err
+	}
+	if err := commands.Init(&rpc.InitRequest{Instance: req.Instance}, nil); err != nil {
+		return nil, err
+	}
+	return &rpc.PlatformUpgradeResponse{}, nil
+}
 
+// platformUpgrade implements the platform upgrade procedure
+func platformUpgrade(
```

Comment:
use a nested function, this way we could remove parameter duplication and could use the defer statement